TagGroup documentation (#3629)
adobe/react-spectrum: packages/@react-spectrum/tag/docs/TagGroup.mdx

  <Item>Adobe AfterEffects</Item>
  <Item>Adobe Illustrator</Item>
  <Item>Adobe Lightroom</Item>
</TagGroup>
```

## Content
TagGroup is a component that allows users to categorize content. Basic usage of TagGroup, seen in the example above, shows the use of a static collection where the contents of the TagGroup are hard coded.
Dynamic collections, as shown below, can be used when the options come from an external data source such as an API, or update over time.

Each item has a unique key defined by the data. In the example below, the `key` of each row element is implicitly defined by the id property of the row object.
See [collections](../react-stately/collections.html#unique-keys) to learn more about keys in dynamic collections.

```tsx example
const items = [
  {id: 1, name: 'Adobe Photoshop'},
  {id: 2, name: 'Adobe XD'},
  {id: 3, name: 'Adobe InDesign'},
  {id: 4, name: 'Adobe AfterEffects'},
  {id: 5, name: 'Adobe Illustrator'},
  {id: 6, name: 'Adobe Lightroom'},
  {id: 7, name: 'Adobe Premiere Pro'},
  {id: 8, name: 'Adobe Fresco'},
  {id: 9, name: 'Adobe Dreamweaver'}
];

<Flex width={500}>
  <TagGroup items={items} aria-label="Dynamic TagGroup items example">

>     {(item) => (
>       <Item key={item.key}>
>         {item.name}
>       </Item>
>     )}

Reviewer: `item.key` doesn't exist. the `id` property will be used automatically, so no `key` prop is needed here.

```suggestion
    {item => <Item>{item.name}</Item>}
```

  </TagGroup>
</Flex>
```

### Internationalization
To internationalize a TagGroup, all text content within the TagGroup should be localized. This includes the `aria-label` provided to the TagGroup if any.
For languages that are read right-to-left (e.g. Hebrew and Arabic), the layout of TagGroup is automatically flipped.

## Labeling
### Accessibility
An `aria-label` must be provided to the TagGroup for accessibility. If the TagGroup is labeled by a separate element, an `aria-labelledby` prop must be provided using the id of the labeling element instead.

## Events

> You can enable tag deletion by providing the `allowsRemoving` prop to the TagGroup. TagGroup will provide the key to be removed to the `onRemove` handler as shown below.

Reviewer: ```suggestion
Removing tags can be enabled by providing the `allowsRemoving` prop to the TagGroup. TagGroup will pass the key of the removed item to the `onRemove` handler as shown below.
```

```tsx example
function Example() {
  let defaultItems = [
    {id: 1, name: 'Adobe Photoshop'},
    {id: 2, name: 'Adobe XD'},
    {id: 3, name: 'Adobe InDesign'},
    {id: 4, name: 'Adobe AfterEffects'},
    {id: 5, name: 'Adobe Illustrator'},
    {id: 6, name: 'Adobe Lightroom'},
    {id: 7, name: 'Adobe Premiere Pro'},
    {id: 8, name: 'Adobe Fresco'},
    {id: 9, name: 'Adobe Dreamweaver'}
  ];

  let [items, setItems] = React.useState(defaultItems);

  let removeItem = (key) => {
    setItems(prevItems => prevItems.filter((item) => key !== item.id));
  };

  return (
    <Flex width={500}>
      <TagGroup items={items} allowsRemoving onRemove={removeItem} aria-label="Removable TagGroup example">

>         {(item) => (
>           <Item key={item.key}>
>             {item.name}
>           </Item>
>         )}

Reviewer: ```suggestion
        {item => <Item>{item.name}</Item>}
```

      </TagGroup>
    </Flex>
  );
}
```

> ## Props

Reviewer: `aria-label` and other AriaLabelingProps seem to be missing here. Maybe they need to be added to the types?

```tsx example
const items = [
  {id: 1, name: 'Adobe Photoshop'},
  {id: 2, name: 'Adobe XD'},
  {id: 3, name: 'Adobe InDesign'},
  {id: 4, name: 'Adobe AfterEffects'},
  {id: 5, name: 'Adobe Illustrator'},
  {id: 6, name: 'Adobe Lightroom'},
  {id: 7, name: 'Adobe Premiere Pro'},
  {id: 8, name: 'Adobe Fresco'},
  {id: 9, name: 'Adobe Dreamweaver'}
];

<Flex width={500}>
  <TagGroup items={items} aria-label="Dynamic TagGroup items example">
    {(item) => (
      <Item key={item.key}>
        {item.name}
      </Item>
    )}
  </TagGroup>
</Flex>
```

### Internationalization
To internationalize a TagGroup, all text content within the TagGroup should be localized. This includes the `aria-label` provided to the TagGroup if any.
For languages that are read right-to-left (e.g. Hebrew and Arabic), the layout of TagGroup is automatically flipped.

## Labeling
### Accessibility
An `aria-label` must be provided to the TagGroup for accessibility. If the TagGroup is labeled by a separate element, an `aria-labelledby` prop must be provided using the id of the labeling element instead.

## Events

> Adding the `allowsRemoving` prop will allow the tags to be removed from the group. The `onRemove` callback is then
> used to handle the remove logic.

Reviewer 2: Mega nit: wording felt a bit awkward and inconsistent from our other docs. Feel free to push back haha
```suggestion
You can enable tag deletion by providing the `allowsRemoving` prop to the TagGroup. TagGroup will provide the key to be removed to the `onRemove` handler as shown below. 
```

As an aside, perhaps we could remove `allowsRemoving` prop and just look for the existence of the `onRemove` prop? `allowsRemoving` feels redundant here IMO since you would always need to provide `onRemove` if you want want removable tags right? No need to address in this PR, just thinking outloud

Reviewer 3: I brought this up with the team, and can't remember what the reasoning was for keeping it. Maybe we can discuss before release.

{/* Copyright 2022 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License. */}

import {Layout} from '@react-spectrum/docs';
export default Layout;

import docs from 'docs:@react-spectrum/tag';
import {HeaderInfo, PropTable, PageDescription} from '@react-spectrum/docs';
import packageData from '@react-spectrum/tag/package.json';

```jsx import
import {Flex} from '@react-spectrum/layout';
import {Item, TagGroup} from '@react-spectrum/tag';
```

---
category: Navigation

> keywords: [TagGroup, navigation]

Reviewer 3: I think perhaps Collections or Status might be a better category since they are intended to categorize content. Not fully sold on either of those though.

Author: I agree. I'm more inclined to go with 'Status'. I put navigation because I saw it under 'Navigation' on the patterns page https://spectrum.corp.adobe.com/page/tag/

Author: @reidbarber should i go ahead and change it to collection or status? do you have a preference?

Reviewer 3: Let's go with Status. We can update it before full release if needed.

---

# TagGroup

<PageDescription>{docs.exports.TagGroup.description}</PageDescription>

<HeaderInfo
  packageData={packageData}

>   componentNames={['Item', 'TagGroup']} />

Reviewer 3: ```suggestion
  componentNames={['Item', 'TagGroup']}
  sourceData={[
    {type: 'Spectrum', url: 'https://spectrum.adobe.com/page/tag/'}
  ]} />
```

    {(item) => (
      <Item key={item.key}>
        {item.name}
      </Item>
    )}
  </TagGroup>
</Flex>
```

You can also disable specific tags by providing an array of keys to TagGroup via the `disabledKeys` prop. This will disable all interactions on disabled tags.

```tsx example
function AppList() {
const defaultItems = [
  {id: 1, name: 'Adobe Photoshop'},
  {id: 2, name: 'Adobe XD'},
  {id: 3, name: 'Adobe InDesign'},
  {id: 4, name: 'Adobe AfterEffects'},
  {id: 5, name: 'Adobe Illustrator'},
  {id: 6, name: 'Adobe Lightroom'},
  {id: 7, name: 'Adobe Premiere Pro'},
  {id: 8, name: 'Adobe Fresco'},
  {id: 9, name: 'Adobe Dreamweaver'}
];

  const [items, setItems] = React.useState(defaultItems);
  const removeItem = (tag) => {
    const updatedItems = items.filter(item => item.name !== tag);
    setItems(updatedItems);
  }

  return (
    <Flex width={500}>
      <TagGroup items={items} disabledKeys={[2,4,5,8]} isRemovable onRemove={removeItem} aria-label="Disable individual tags example">
        {(item) => (
          <Item key={item.key}>
            {item.name}
          </Item>
        )}
      </TagGroup>
    </Flex>
  );
}

```

> ### Removing Tags

Reviewer 3: ``` suggestion
### Removing tags
```
Small nit, to match other docs

Adding the `isRemovable` prop will allow the tags to be removed from the group. The `onRemove` callback is then
used to handle the remove logic.

```tsx example
function AppList() {
const defaultItems = [
  {id: 1, name: 'Adobe Photoshop'},
  {id: 2, name: 'Adobe XD'},
  {id: 3, name: 'Adobe InDesign'},
  {id: 4, name: 'Adobe AfterEffects'},
  {id: 5, name: 'Adobe Illustrator'},
  {id: 6, name: 'Adobe Lightroom'},
  {id: 7, name: 'Adobe Premiere Pro'},
  {id: 8, name: 'Adobe Fresco'},
  {id: 9, name: 'Adobe Dreamweaver'}
];

  const [items, setItems] = React.useState(defaultItems);
  const removeItem = (tag) => {
    const updatedItems = items.filter(item => item.name !== tag);
    setItems(updatedItems);
  }

  return (
    <Flex width={500}>
      <TagGroup items={items} isRemovable onRemove={removeItem} aria-label="Removable TagGroup example">
        {(item) => (
          <Item key={item.key}>
            {item.name}
          </Item>
        )}
      </TagGroup>
    </Flex>
  );
}

```

> ### Overflow

Reviewer 3: IMO, we could consider omitting this section since it's the default behavior

If the length of the text is longer than the width of the component, the text will truncate with an ellipsis.

```tsx example
const items = [
  {id: 1, name: 'Adobe Photoshop'},
  {id: 2, name: 'Adobe XD'},
  {id: 3, name: 'Adobe InDesign'},
  {id: 4, name: 'Adobe AfterEffects'},
  {id: 5, name: 'Adobe Illustrator'},
  {id: 6, name: 'Adobe Lightroom'},
  {id: 7, name: 'Adobe Premiere Pro'},
  {id: 8, name: 'Adobe Fresco'},
  {id: 9, name: 'Adobe Dreamweaver'}
];

<Flex width={100}>
  <TagGroup items={items} aria-label="Overflow TagGroup example">
    {(item) => (
      <Item key={item.key}>
        {item.name}
      </Item>
    )}
  </TagGroup>
</Flex>
```

## Props

> <PropTable component={{

Reviewer 3: I'm seeing `onSelectionChange` in the table, but I don't think that should be there. Also it looks like `disabledKeys`, `isDisabled`, `isRemovable`, and `onRemove` all needs descriptions.
